### PR TITLE
Add boot sequence information

### PR DESCRIPTION
**Pull request by dpetrisko**
This PR adds boot sequence information and cleans up some platform guide info which was out of date

### CODE REVIEW

File: docs/debug_boot.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+The first method of boot involves a host tether which sets up BlackParrot for execution. The BlackParrot NBF boot happens in several steps. An NBF (network boot format) loader in the testbench takes in an NBF file, and converts it into a series of I/O writes into the core. This I/O fills the DRAM, cce microcode, and sets configuration registers within BlackParrot. Usage of nbf.py is shown below:
+
+
+    usage: nbf.py [-h] [--ncpus NCPUS] [--ucode ucode.mem] [--mem prog.mem]
+              [--config] [--checkpoint sample.nbf] [--skip_zeros]
+              [--addr_width ADDR_WIDTH]
+
+The default NPF boot sequence is as follows (waveforms shown from executing the "hello world" program):
+- reset is lowered. This must be lowered for 10 or more cycles. reset is raised.
+- The freeze register is set. This prevents BlackParrot from fetching or executing instructions.
```
Comment (taylor-bsg) [inline docs/debug_boot.md:24]: By default, the freeze register is set?

File: docs/debug_boot.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+## Boot Process (Method 1: NBF)
+
+The first method of boot involves a host tether which sets up BlackParrot for execution. The BlackParrot NBF boot happens in several steps. An NBF (network boot format) loader in the testbench takes in an NBF file, and converts it into a series of I/O writes into the core. This I/O fills the DRAM, cce microcode, and sets configuration registers within BlackParrot. Usage of nbf.py is shown below:
+
+
+    usage: nbf.py [-h] [--ncpus NCPUS] [--ucode ucode.mem] [--mem prog.mem]
+              [--config] [--checkpoint sample.nbf] [--skip_zeros]
+              [--addr_width ADDR_WIDTH]
+
+The default NPF boot sequence is as follows (waveforms shown from executing the "hello world" program):
```
Comment (taylor-bsg) [inline docs/debug_boot.md:22]: link to hello world program?

File: docs/debug_boot.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+- reset is lowered. This must be lowered for 10 or more cycles. reset is raised.
+- The freeze register is set. This prevents BlackParrot from fetching or executing instructions.
+- I/O writes begin happening. There should be a single response for each request. These should look like the following:
+
+![I/O Writes](debug_io.png)
+
```
Comment (taylor-bsg) [inline docs/debug_boot.md:28]: The waveform diagrams are great, but can we provide example fully qualified paths? Would a DVE or other viewer configure file provide these?

File: docs/debug_boot.md (hunk 1 above)
Comment (dpetrisko) [reply docs/debug_boot.md:24]: The NBF loader also redundantly set it

File: docs/debug_boot.md (hunk 2 above)
Comment (dpetrisko) [reply docs/debug_boot.md:22]: It's in the same place as all other programs, having redundant links just increases the chance they go out of date

**Pull request merged**